src/app/components/control-bar/control-bar.component.ts

```diff
@@ -41,6 +42,18 @@ export class ControlBarComponent implements OnInit {
       }
     });
 
+    combineLatest(this.notificationsService.loadingOverlayData,
+                  this.notificationsService.loadingPointCloudData,
+                  this.notificationsService.loadingFeatureData)
+      .subscribe(([loadingOverlay, loadingPointCloud, loadingFeature]) => {
+        // They are running
+        if (!loadingOverlay && !loadingPointCloud && !loadingFeature) {
```

Comment:
suggestion/nitpick for readability (feel free to ignore): 
```

if (loadingOverlay || loadingPointCloud || loadingFeature) {
   this.loadingData = true
```

Reply:
@nathanfranklin I've changed it to.
``` javascript
if (!(loadingOverlay || loadingPointCloud || loadingFeature)) {
   this.loadingData = true
```